edpop-record-ontology.ttl

```diff
@@ -212,3 +228,41 @@ edpoprec:physicalDescription a rdf:property ;
                              rdfs:domain edpoprec:BibliographicalRecord ;
                              rdfs:range edpoprec:Field .
 
+### FIELDS SPECIFIC FOR BIOGRAPHICAL RECORDS
+
+edpoprec:name a rdf:property ;
+              skos:prefLabel "Name"@en ;
+              rdfs:domain edpoprec:BiographicalRecord ;
+              rdfs:range edpoprec:Field .
+
+edpoprec:variantName a rdf:property ;
+              skos:prefLabel "Name"@en ;
+              rdfs:domain edpoprec:BiographicalRecord ;
+              rdfs:range edpoprec:Field .
```

Comment:
It may be cleaner to add variants as `edpoprec:name`. If there is some crucial distinction between `name` and `variantName`, you should explain it here.

Reply:
Hm, I don't know whether "alternative name" is semantically meaningful (or if other catalogues just include it for technical reasons), but if so, that distinction makes sense.

Some more thoughts:

- I don't know if there are specific reasons for needing a default name? Perhaps it's not necessary. As long as your interface can display multiple names, that can be fine too.
- I would not use a `default` subfield - setting this to true for one and only one property within the range of an entity is hard to enforce.
- An alternative that is least easier to validate: you could have `:name` point to multiple objects and add a functional property `:defaultName`. Thus, an entity might have five `:name` relations, one of which is also its `:defaultName`.

Reply:
The distinction will be important, because some records have more than ten alternative names/titles. That happens especially with CERL Thesaurus as this database tries to record all variants of a name across Europe. But they have selected the most common or 'normal' one as the main name. We will need that as well; for instance, the entry for Vondel has over 50 variant names, of which the first one is Йост ван ден Вондел, and I don't think we want to show this one first :)

Reply:
If you add a separate (functional) property `defaultName`, I don't really see the advantage over having separate normal properties... I also think that it will make the implementation more complicated.

It will be problematic indeed to enforce that there is only one field marked as `default`. Not sure if that is necessary, but certainly not very clean either.

Reply:
This is how it was solved in BIBFRAME (the linked data successor of Marc21): https://www.loc.gov/bibframe/docs/pdf/bf2-titles-march2017.pdf

So a work has only one title property, `title`, of class `Title`. Since it is RDF it can of course have multiple values. But `Title` has various subclasses, such as `AlternativeTitle`, `AbbreviatedTitle` and also `KeyTitle`, so that the distinction can still be made.

This could work in our case as well; then we would have to create subclasses of `Field` both in the ontology and in the API, and we can simply follow the semantics of the catalogues to decide whether something is 'alternative' or not.

What about this solution?

Reply:
I would keep it simple and go with `name` and `alternativeName`. That is also conventional, e.g. SKOS does that with labels. You can also add a `preferredName` in the mix, if you think this is useful. All those properties can have multiple values, for example in order to support multiple languages.

`variantName` sounds like it is the name of a variant of something, so I think that name would be less suitable. But the principle seems correct.